### PR TITLE
[ISSUE-3453]: defaults choices must be an array sf2.8

### LINKED ISSUE #3452: [SF 2.8] Argument 1 passed to Symfony\Component\Form\Extension\Core\Type\ChoiceType::normalizeLegacyChoices() must be of the type array, null given
On Symfony 2.8 only, with the following form config:

```php
->add('groups', 'sonata_type_model', ['required' => false, 'expanded' => true, 'multiple' => true])
```

I got the following error:

```
Catchable Fatal Error: Argument 1 passed to Symfony\Component\Form\Extension\Core\Type\ChoiceType::normalizeLegacyChoices() must be of the type array, null given, called in /home/projects/project/vendor/symfony/symfony/src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php on line 266 and defined
```

Seems to be related to this config line: https://github.com/sonata-project/SonataAdminBundle/blob/09cb3bf45b4c662059c668173f3c85234d825767/Form/Type/ModelType.php#L105

If i replace `null` by `array()`, no error thrown but my choice form is empty.

Don't know wait if it's a Symfony or a Sonata issue, some investigation is needed.

### LINKED ISSUE #3453: sonata_type_model compatible with Symfony 2.8
normalizeLegacyChoices Forces the use of an array in Defaults 'choices', this will be only supported until version 3.0

### PR DESCRIPTION
**Pull request by Trip09**
Fixes #3452.
Fixes #3453.

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
SonataCI seems to be broken.

@rande Could we have a link on the PR boxes? It' hard for us and contributors to see what happened.

**Comment by Soullivaneuh** [comment]
@Trip09 Could you please add related unit test that prove the error and the fix?

**Comment by Soullivaneuh** [comment]
Manually tested, this is OK. :+1: 

Waitin @rande for SonataCI being stuck.

File: Form/ChoiceList/ModelChoiceList.php [hunk 1]
```diff
@@ -128,7 +128,7 @@ public function __construct(ModelManagerInterface $modelManager, $class, $proper
      */
     protected function load($choices)
     {
-        if (is_array($choices)) {
+        if (is_array($choices) && 0 < count($choices)) {
```
Comment (Soullivaneuh) [inline Form/ChoiceList/ModelChoiceList.php]:
```php
count($choices) > 0
```

In Symfony, yoda condition are applied only for equal statements.

**Comment by Trip09** [comment]
@Soullivaneuh tests add when I run phpunit the tests are passing there are only *Remaining deprecation notices (15)* so I dunno why SonataCI is broken...

**Comment by Padam87** [comment]
2nd confirmation. Works fine.

**Comment by Soullivaneuh** [comment]
@rande Is actually incorporating your PR for 3.0 compatibility.

**Comment by OskarStark** [comment]
is this still needed @rande ?

**Comment by karousn** [comment]
i have the some problem with symfony 2.8 when i try to add some entity in classification angle

**Comment by Trip09** [comment]
some ideas about merging this? @Soullivaneuh @OskarStark @rande

**Comment by rande** [comment]
@Trip09 sorry I didn't take this PR. Can you rebase it ? So unit tests should pass.

Thanks!

**Comment by Trip09** [comment]
@rande SonataCI is faling because Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList is deprecated and will be removed in sf3 the same will happen in travis when testing for sf3, if I change this PR to five support to sf3 the branch master no longer be compatible with sf < 2.7

**Comment by rande** [comment]
Just skip the test, something like this: https://github.com/sonata-project/SonataAdminBundle/blob/d9c85b2a76f72e23b861e33d0f67abdc4ea2c963/Tests/Form/DataTransformer/LegacyModelsToArrayTransformerTest.php#L33-L35

**Comment by Trip09** [comment]
@rande nice! all passing tx

**Comment by rande** [comment]
@Trip09 thanks!

**Comment by paquitodev** [comment]
Also I am seeing weird things like the one on the pic, that doesn't appears to me:

**Comment by mediamonks-robert** [comment]
Can anyone post their composer.json to make Sonata admin + page work with 2.8?

**Comment by Trip09** [comment]
@mediamonks-robert you can't copy paste ( there are private repos) but you can have an idea http://pastebin.com/MXTHBFDA

**Comment by Trip09** [comment]
@paquitodev not related can't reproduce your issue

**Comment by Padam87** [comment]
Composer update might be hard (maybe even impossible), depending on your dependendency graph. I ended up forking 2.3.7 and cherry picking this commit over to a new branch...

```
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/Padam87/SonataAdminBundle.git"
        }
    ],

    ...

    "sonata-project/admin-bundle": "dev-2.8-comp",
```

Feel free to use it, I wont mess with it for a while.

**Comment by paquitodev** [comment]
@Padam87 thanks I will use as well ;)

**Comment by paquitodev** [comment]
@Padam87 I am running this issue with your solution:

    Problem 1
      - Installation request for sonata-project/admin-bundle dev-2.8-comp -> satisfiable by sonata-project/admin-bundle[dev-2.8-comp].
      - sonata-project/admin-bundle dev-2.8-comp requires sensio/generator-bundle ~2.3 -> no matching package found.

This is my `composer.json`:

    "require": {
        "php": ">=5.3.9",
        "symfony/symfony": "2.8.*",
        "doctrine/orm": "^2.4.8",
        "doctrine/doctrine-bundle": "~1.4",
        "symfony/swiftmailer-bundle": "~2.3",
        "symfony/monolog-bundle": "~2.4",
        "sensio/distribution-bundle": "~5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "~2.0",
        "symfony/security-acl": "^2.7",
        "knplabs/knp-menu-bundle": "dev-master",
        "sonata-project/admin-bundle": "dev-2.8-comp",
        "sonata-project/doctrine-orm-admin-bundle": "^2.3",
        "sonata-project/block-bundle": "^2.3",
        "sonata-project/user-bundle": "^2.2",
        "sonata-project/easy-extends-bundle": "^2.1",
        "avanzu/admin-theme-bundle": "^1.3",
        "symfony/assetic-bundle": "^2.7",
        "kevintweber/ktw-database-menu-bundle": "~0.5",
        "jms/serializer-bundle": "0.13.*",
        "stof/doctrine-extensions-bundle": "^1.2",
        "friendsofsymfony/user-bundle": "1.3.x-dev",
        "friendsofsymfony/rest-bundle": "~1.5",
        "nelmio/api-doc-bundle": "2.8.*@dev",
        "mandrill/mandrill": "dev-master",
        "nzo/url-encryptor-bundle": "~1.0",
        "knpuniversity/guard-bundle": "~0.1@dev",
        "halleck45/phpmetrics-collector-bundle": "*"
    },
    "require-dev": {
        "sensio/generator-bundle": "~3.0",
        "symfony/phpunit-bridge": "~2.7"
    },

What's wrong?

**Comment by Padam87** [comment]
@paquitodev try `dev-2.8-comp@dev`

You will probably need to change your requirement of the block bundle, ^2.3 is too high for the latest stable release of the admin bundle (https://github.com/sonata-project/SonataAdminBundle/blob/2.3.7/composer.json#L35)

**Comment by paquitodev** [comment]
@Padam87 same issues, I will wait until Sonata team fix this issues to continue my development, I am stopped because of this anyway thanks for your help

**Comment by meshenka** [comment]
Got same issue with Sf 2.7.8 + SonataAdmin 2.3.7
can it be backported to 2.3 branch ?

**Pull request merged**